src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -2253,6 +2253,9 @@ ITypeSymbol ITypeSymbolInternal.GetITypeSymbol()
             return GetITypeSymbol(DefaultNullableAnnotation);
         }
 
+        // PROTOTYPE(record-structs): rename to IsRecordClass?
         internal abstract bool IsRecord { get; }
+
+        internal abstract bool IsRecordStruct { get; }
```

Comment:
:memo: Alternatively, we could re-use and extend `IsRecord`. Some places that previously checked `IsRecord` would need to be updated to also check `IsReferenceType`. This probably should be discussed more in context of public API we want to expose (tracked in test plan).

Reply:
If it is convenient to have `IsRecordStruct`, it might make sense to have the following set of properties

```cs
// new property
internal abstract bool IsRecord { get; }

// current IsRecord property renamed (with usages audited to determine what property they should be changed to)
internal bool IsRecordClass => IsRecord && TypeKind == TypeKind.Class;

internal bool IsRecordStruct => IsRecord && TypeKind == TypeKind.Struct;
```

Reply:
Right, that's what I was considering. I have a follow-up in test plan and also PROTOTYPE above to track.

---
In reply to: [588778787](https://github.com/dotnet/roslyn/pull/51494#discussion_r588778787) [](ancestors = 588778787)